src/commands/workspace/list.ts

```diff
@@ -19,30 +23,53 @@ export default class List extends Command {
     help: flags.help({ char: 'h' }),
     chenamespace: cheNamespace,
     'access-token': accessToken,
+    'skip-count': flags.integer({
+      description: 'The number of workspaces to skip',
+      default: 0
+    }),
+    'max-items': flags.integer({
+      description: 'The maximum number of workspaces to return',
+      default: 30
+    }),
     'listr-renderer': listrRenderer
   }
 
   async run() {
     const { flags } = this.parse(List)
-    const Listr = require('listr')
-    const notifier = require('node-notifier')
-    const tasks = new Listr([
-      { title: 'Verify if we can access Kubernetes API', skip: this.warn('Not implemented yet') },
-      { title: 'Verify if Eclipse Che is running', skip: this.warn('Not implemented yet') },
-      { title: 'Get Workspaces', skip: this.warn('Not implemented yet') },
-    ], { renderer: flags['listr-renderer'] as any })
-
-    // Use https://github.com/oclif/cli-ux/tree/supertable#clitable to dispalay:
-    //  - workspace id
-    //  - workspace state
-    //  - workspace creation date ?
-    //  - workspace stack ?
-
-    await tasks.run()
-
-    notifier.notify({
-      title: 'chectl',
-      message: 'Command workspace:list has completed.'
+    const ctx: any = {}
+
+    const apiTasks = new ApiTasks()
+    const cheTasks = new CheTasks(flags)
+    const tasks = new Listrq(undefined, { renderer: flags['listr-renderer'] as any })
+
+    tasks.add(apiTasks.testApiTasks(flags, this))
+    tasks.add(cheTasks.verifyCheNamespaceExistsTask(flags, this))
+    tasks.add(cheTasks.retrieveEclipseCheUrl(flags))
+    tasks.add(cheTasks.checkEclipseCheStatus())
+    tasks.add({
+      title: 'Get workspaces',
+      task: async (ctx, task) => {
+        const cheHelper = new CheHelper(flags)
+        ctx.workspaces = await cheHelper.getWorkspaces(ctx.cheURL, flags['skip-count'], flags['max-items'], flags['access-token'])
+        task.title = `${task.title}... done`
+      }
     })
+
+    try {
+      await tasks.run(ctx)
+      this.printWorkspaces(ctx.workspaces)
+    } catch (error) {
+      this.error(error)
+    }
+  }
+
+  private printWorkspaces(workspaces: any): void {
+    this.log('Id\tName\tNamespace\tStatus\tLast modified')
```

Comment:
I think we may hardcode right number of tabs here. Then we'll have table-like header.

Reply:
It depends on length of data

Reply:
Well, we may predict that workspace id is always as long as `workspaceadm0rrdlo2kt9hj9` and the same for other fields except workspace name. Here we can either truncate it or move to the last column.

Reply:
You never know the namespace length.
In general case it is needed to implement some complex logic.
I would keep as is for know